.github/workflows/build-win-native.yml

```diff
@@ -0,0 +1,105 @@
+        uses: msys2/setup-msys2@v2
+        with:
+          path-type: inherit
+          install: >-
+            procps
+            mingw-w64-x86_64-protobuf
+
+      - name: "Install: Rust toolchain"
+        uses: dsherret/rust-toolchain-file@v1
+
+      - name: "Install: cargo-nextest"
+        run: |
+          $tmp = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'zip' } -PassThru
+          Invoke-WebRequest -OutFile $tmp https://get.nexte.st/latest/windows
+          $outputDir = if ($Env:CARGO_HOME) { Join-Path $Env:CARGO_HOME "bin" } else { "~/.cargo/bin" }
+          $tmp | Expand-Archive -DestinationPath $outputDir -Force
+          $tmp | Remove-Item
+        shell: powershell
+
+      - name: "Install: cargo-hack"
+        run: |
+          $tmp = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar.gz' } -PassThru
+          Invoke-WebRequest -OutFile $tmp https://github.com/taiki-e/cargo-hack/releases/latest/download/cargo-hack-x86_64-pc-windows-msvc.tar.gz
+          $outputDir = if ($Env:CARGO_HOME) { Join-Path $Env:CARGO_HOME "bin" } else { Resolve-Path "~/.cargo/bin" }
+          tar zxf $tmp -C $outputDir
+          $tmp | Remove-Item
+        shell: powershell
+
+      - name: "Install: binaryen"
+        run: |
+          $tmp = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar.gz' } -PassThru
+          Invoke-WebRequest -OutFile $tmp "https://github.com/WebAssembly/binaryen/releases/download/$Env:BINARYEN_VERSION/binaryen-$Env:BINARYEN_VERSION-x86_64-windows.tar.gz"
+          $outputDir = if ($Env:CARGO_HOME) { Join-Path $Env:CARGO_HOME "bin" } else { "~/.cargo/bin" }
+          tar zxf $tmp
+          cp "./binaryen-$Env:BINARYEN_VERSION/bin/*" $outputDir
+          Remove-Item -Recurse "./binaryen-$Env:BINARYEN_VERSION"
+          $tmp | Remove-Item
+        shell: powershell
+
+      - name: "Install: wabt"
+        run: |
+          $tmp = New-TemporaryFile | Rename-Item -NewName { $_ -replace 'tmp$', 'tar.gz' } -PassThru
+          Invoke-WebRequest -OutFile $tmp https://github.com/WebAssembly/wabt/releases/download/1.0.29/wabt-1.0.29-windows.tar.gz
+          $outputDir = if ($Env:CARGO_HOME) { Join-Path $Env:CARGO_HOME "bin" } else { "~/.cargo/bin" }
+          tar zxf $tmp
+          cp ./wabt-1.0.29/bin/* $outputDir
+          Remove-Item -Recurse ./wabt-1.0.29
+          $tmp | Remove-Item
+        shell: powershell
+
+      - name: "Build: Node"
+        run: ./scripts/gear.sh build node --release --locked
+
+      - name: "Check: Stack height limit"
+        # no need to run check twice
+        if: ${{ matrix.profile == 'release' }}
+        run: cargo run -p calc-stack-height --release --locked
+
+      - name: "Test: gsdk tests"
+        run: ./scripts/gear.sh test gsdk ${{ matrix.profile_flags }}
+
+      - name: "Test: Client tests"
+        run: ./scripts/gear.sh test client ${{ matrix.profile_flags }}
+
+      - name: "Test: Lazy pages"
```

Comment:
please rename here and in other places. maybe "workspace tests" or "pallet, lazy-pages and runtime tests". Actually I don't know whether we don't run all workspace tests here

Reply:
I don't see reason to rename the step as its purpose is to test lazy-pages despite the step runs all of these tests

Reply:
So, it leads to misleading when some pallet tests will fail, because it won't be actually lazy-pages problem

Reply:
Why we don't call `gear.sh test gear` here ?